glean-core/tests/labeled.rs

```diff
@@ -247,6 +247,7 @@ fn dynamic_labels_regex_mimsatch() {
         "1.not_fine",
         "this.$isnotfine",
         "-.not_fine",
+        "this.is_not_fine.2",
```

Comment:
Why is this not fine? Is this different from [this](https://github.com/mozilla/glean/pull/668/files#diff-15c101f06a477efbf7196d7571cfe05cR27)?

Reply:
BQ may put these labels into a structure, and the names of each element can't begin with a number.  Fortunately, no one is doing this yet.

Reply:
Ouch :| So would `this.is_not_fine.2` be fine but not `2this.is_not_fine` ?

Reply:
Neither is fine.